pkg/query-service/rules/manager.go

```diff
@@ -233,7 +253,24 @@ func (m *Manager) EditRule(ruleStr string, id string) error {
 	}
 
 	if !m.opts.DisableRules {
-		return m.syncRuleStateWithTask(taskName, parsedRule)
+		err = m.syncRuleStateWithTask(taskName, parsedRule)
+		if err != nil {
+			return err
+		}
+	}
+
+	// update feature usage if the current rule is not a trace or log query builder
+	if !checkIfTraceOrLogQB(&currentRule.PostableRule) {
+		err = m.updateFeatureUsage(parsedRule, 1)
+		if err != nil {
+			zap.S().Errorf("error updating feature usage: %v", err)
+		}
+		// update feature usage if the new rule is not a trace or log query builder and the current rule is
+	} else if !checkIfTraceOrLogQB(parsedRule) {
+		err = m.updateFeatureUsage(&currentRule.PostableRule, -1)
+		if err != nil {
+			zap.S().Errorf("error updating feature usage: %v", err)
+		}
```

Comment:
Isn't this updating the usage when the source is metrics? Or am I misunderstanding something?

Reply:
I have explained it here: https://github.com/SigNoz/signoz/pull/2661/files#r1195907615
If still unclear please tell me scenario in which this logic doesn't work.

Reply:
I will pull and take a look at it closely.

Reply:
The different internal implementations of `updateFeatureUsage` in dashboards and rules confused it. The `updateFeatureUsage` in the dashboard updates it directly; however, The `updateFeatureUsage` in the rules doesn't do the same, but only when the `isTraceOrLogQB` is true. I read this EditRule part assuming they have similar logic hence the questions. If you could find a way to change `updateFeatureUsage` in rules not to have further conditional logic, that would be great for future readers.

Reply:
Got it.